### PR TITLE
fix: make proto targets in golang/cosmos

### LINKED ISSUE #4063: Protobuf utilities in golang/cosmos are broken
## Describe the bug

The protobuf related `make` targets in `golang/cosmos` are in various states of broken.

In no particular order:

- `make proto-lint` complains about a missing `gogo.proto`, despite the fact that it's in `third_party`
- the `protoc-gen-grpc-gateway` tools must be installed manually (undocumented)
- `make proto-gen` does not work
- several other issues

## To Reproduce
See above

## Expected behavior
The `make` targets should download the files and tools as necessary to work.

## Platform Environment
MacOS

### PR DESCRIPTION
**Pull request by JimLarson**
closes: #4063

## Description

- Make grpc-gateway part of the proto tools and update the version.
- Update buf configuration
- Correct go package for `any.proto`.

### Security Considerations

None.

### Documentation Considerations

None.

### Testing Considerations

Can do a breaking check as part of CI (once it's passing).

It would be good to exercise all these tools as part of CI to make sure they don't break.

### CODE REVIEW

File: golang/cosmos/third_party/proto/google/protobuf/any.proto [hunk 1]
```diff
@@ -35,7 +35,7 @@ package google.protobuf;
 import "gogoproto/gogo.proto";
 
 option csharp_namespace = "Google.Protobuf.WellKnownTypes";
-option go_package = "types";
+option go_package = "github.com/cosmos/cosmos-sdk/codec/types";
```
Comment (michaelfig) [inline golang/cosmos/third_party/proto/google/protobuf/any.proto]: I don't think this affects any of our code (given the explicit remap of `google/protobuf/any.proto` in the `protogen.sh`), but I would prefer not to diverge unnecessarily from `$GOPATH/include/google/protobuf/any.proto`.  I think this should be:

```suggestion
option go_package = "github.com/golang/protobuf/ptypes/any";
```

File: golang/cosmos/tools/tools.go [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// +build tools
+
```
Comment (michaelfig) [inline golang/cosmos/tools/tools.go:4]:
```suggestion
// This file exists only to make references to tool modules we need.
```

File: golang/cosmos/scripts/protocgen.sh [hunk 3]
```diff
@@ -9,7 +9,7 @@ for dir in $proto_dirs; do
   -I third_party/proto \
   --gocosmos_out=plugins=interfacetype+grpc,\
 Mgoogle/protobuf/any.proto=github.com/cosmos/cosmos-sdk/codec/types:. \
-  --grpc-gateway_out=logtostderr=true,allow_colon_final_segments=true:. \
+  --grpc-gateway_out=logtostderr=true:. \
```
Comment (michaelfig) [inline golang/cosmos/scripts/protocgen.sh]: I was curious about this change, then I found the v2 migration docs.  We are still using v1.x, so please revert this change and rerun `make proto-gen`.

To preserve what you learned, could you please instead add a comment like:

```sh
# NOTE: when migrating to grpc-gateway v2, we will need to remove the
# allow_colon_final_segments=true
# as per https://grpc-ecosystem.github.io/grpc-gateway/docs/development/grpc-gateway_v2_migration_guide/#withlastmatchwins-and-allow_colon_final_segmentstrue-is-now-default-behaviour
```

File: golang/cosmos/x/swingset/types/query.pb.gw.go [hunk 4]
```diff
@@ -534,15 +551,15 @@ func RegisterQueryHandlerClient(ctx context.Context, mux *runtime.ServeMux, clie
...
+	pattern_Query_Mailbox_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2, 1, 0, 4, 1, 5, 3}, []string{"agoric", "swingset", "mailbox", "peer"}, "", runtime.AssumeColonVerbOpt(true)))
 
-	pattern_Query_Storage_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2, 2, 3, 1, 0, 4, 1, 5, 4}, []string{"agoric", "swingset", "storage", "data", "path"}, "", runtime.AssumeColonVerbOpt(false)))
+	pattern_Query_Storage_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2, 2, 3, 1, 0, 4, 1, 5, 4}, []string{"agoric", "swingset", "storage", "data", "path"}, "", runtime.AssumeColonVerbOpt(true)))
 
-	pattern_Query_Keys_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2, 2, 3, 1, 0, 4, 1, 5, 4}, []string{"agoric", "swingset", "storage", "keys", "path"}, "", runtime.AssumeColonVerbOpt(false)))
+	pattern_Query_Keys_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2, 2, 3, 1, 0, 4, 1, 5, 4}, []string{"agoric", "swingset", "storage", "keys", "path"}, "", runtime.AssumeColonVerbOpt(true)))
```
Comment (michaelfig) [inline golang/cosmos/x/swingset/types/query.pb.gw.go]: I would not expect these lines to change.

File: golang/cosmos/x/vbank/types/query.pb.gw.go [hunk 5]
```diff
@@ -198,9 +206,9 @@ func RegisterQueryHandlerClient(ctx context.Context, mux *runtime.ServeMux, clie
 }
 
 var (
-	pattern_Query_Params_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2}, []string{"agoric", "vbank", "params"}, "", runtime.AssumeColonVerbOpt(false)))
+	pattern_Query_Params_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2}, []string{"agoric", "vbank", "params"}, "", runtime.AssumeColonVerbOpt(true)))
 
-	pattern_Query_State_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2}, []string{"agoric", "vbank", "state"}, "", runtime.AssumeColonVerbOpt(false)))
+	pattern_Query_State_0 = runtime.MustPattern(runtime.NewPattern(1, []int{2, 0, 2, 1, 2, 2}, []string{"agoric", "vbank", "state"}, "", runtime.AssumeColonVerbOpt(true)))
```
Comment (michaelfig) [inline golang/cosmos/x/vbank/types/query.pb.gw.go]: Same, I would expect no change.

**Review by michaelfig — COMMENTED** [review]
Wow!  Thanks for all this work.

I have just a few questions, and when you address them to your satisfaction, please ping me for another review and I will approve.

File: golang/cosmos/scripts/protocgen.sh (hunk 3 above)
Comment (JimLarson) [reply golang/cosmos/scripts/protocgen.sh]: Done and done.

File: golang/cosmos/tools/tools.go (hunk 2 above)
Comment (JimLarson) [reply golang/cosmos/tools/tools.go:4]: Done.

File: golang/cosmos/third_party/proto/google/protobuf/any.proto (hunk 1 above)
Comment (JimLarson) [reply golang/cosmos/third_party/proto/google/protobuf/any.proto]: Done.

File: golang/cosmos/x/swingset/types/query.pb.gw.go (hunk 4 above)
Comment (JimLarson) [reply golang/cosmos/x/swingset/types/query.pb.gw.go]: And now they don't, given the fix above.

File: golang/cosmos/x/vbank/types/query.pb.gw.go (hunk 5 above)
Comment (JimLarson) [reply golang/cosmos/x/vbank/types/query.pb.gw.go]: Same reply.

**Comment by JimLarson** [comment]
Addressed review comments - PTAL

**Review by michaelfig — APPROVED** [review]
Terrific!
